src/app/project/project-details.page.ts

```diff
@@ -0,0 +1,23 @@
+import {  Component, inject } from '@angular/core';
+import {  NavigationEnd, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-project',
+  templateUrl: './project-details.page.html',
+  styleUrls: ['./project-details.page.scss'],
+})
+export class ProjectDetailsPage  {
+  router: Router;
+  projectData:any;
+    constructor() {
+        this.router = inject(Router);
+      this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
```

Comment:
Can you move this subscription logic outside of the constructor function into ngOnInit life cycle hook? Its recommended to use construction function for dependency injection and component logic should be handled in hooks or custom methods

Reply:
done